### ISSUE TITLE
Only block 3rd-party requests if the request domain isn't in the same "group" as the origin domain 

### ISSUE THREAD

**Issue by groovecoder** [mozilla/blok#1]

**Comment by fmarier** [comment]
This is based on this list: https://github.com/mozilla-services/shavar-prod-lists/blob/master/disconnect-entitylist.json

The entity list is transformed into a pairwise allowlist like this:

```
foreach company:
    foreach property:
        foreach resource:
            addRule("if on property, then don't block resource")
```

and when we are about to block a resource, we check if there is a rule in the pairwise allowlist for that particular (property, resource) pair.

**Comment by groovecoder** [comment]
So, for example, the entitylist has:
```
"Google": {
  "properties": [
    "google.com",
    "ingress.com"
    ...
  ],
  "resources": [
    "google.com",
    "doubleclick.net",
    ...
  ]
},
...
"AT&T": {
  "properties": [
    "att.com",
    "yp.com"
  ],
  "resources": [
    "att.com",
    "yp.com"
  ]
}
...
```
So, in plain English, before we block a request to `doubleclick.net`, we should check if the origin is one of `Google.properties` ? I.e.,

* origin: `google.com` + request: `google.com` = allow
* origin: `ingress.com` + request: `doubleclick.net` = allow
* origin: `ingress.com` + request: `yp.com` = block
* origin: `att.com` + request: `yp.com` = allow
* origin: `att.com` + request: `doubleclick.net` = block
?

**Comment by groovecoder** [comment]
@fmarier: I have code for this now. Do you know a good test case I can use to make sure it's working? I.e., a property page that makes a resource request to a domain in the same entity?

**Comment by fmarier** [comment]
> Do you know a good test case I can use to make sure it's working? I.e., a property page that makes a resource request to a domain in the same entity?

I've used https://youtube.com for this in the past.

**Comment by groovecoder** [comment]
Looks like youtube.com makes requests to doubleclick.net? So the add-on should **NOT** block those requests?

**Comment by fmarier** [comment]
> Looks like youtube.com makes requests to doubleclick.net? So the add-on should NOT block those requests?

That's right, they shouldn't be blocked because both `youtube.com` and `doubleclick.net` are Google's.

**Comment by groovecoder** [comment]
Fixed in https://github.com/mozilla/blok/pull/23
